### PR TITLE
 bpo-36231: no "proper" header files on macOS 10.14 Mojave

### PR DESCRIPTION
**Pull request by dimpase**
https://bugs.python.org/issue36231

### CODE REVIEW

File: aclocal.m4 [hunk 1]
```diff
@@ -1,6 +1,6 @@
-# generated automatically by aclocal 1.16.1 -*- Autoconf -*-
+# generated automatically by aclocal 1.15 -*- Autoconf -*-
```
Comment (embray) [inline aclocal.m4]: Might want to remove these aclocal.m4 changes since they seem to be irrelevant.

File: configure [hunk 2]
```diff
@@ -1436,6 +1447,7 @@ Fine tuning of the installation directories:
   --sysconfdir=DIR        read-only single-machine data [PREFIX/etc]
   --sharedstatedir=DIR    modifiable architecture-independent data [PREFIX/com]
   --localstatedir=DIR     modifiable single-machine data [PREFIX/var]
+  --runstatedir=DIR       modifiable per-process data [LOCALSTATEDIR/run]
```
Comment (embray) [inline configure]: Ditto here.  The only diff in the `configure` that's relevant is the last one beginning here: https://github.com/python/cpython/pull/12825/files#diff-e2d5a00791bce9a01f99bc6fd613a39dR7369

File: aclocal.m4 (hunk 1 above)
Comment (dimpase) [reply aclocal.m4]: these are autotools-generated, they just not need to be reviewed.
(I guess the only reason cpython keeps these in the repo is convenience for people working
on autotools-less systems, for one way or another).

File: configure (hunk 2 above)
Comment (dimpase) [reply configure]: same as above: these are autotools-generated, they just not need to be reviewed.

Comment (embray) [reply configure]: I know: My point is you can drop this changes from the PR because they're not relevant even as autogenerated autotools stuff.  Better to just keep the relevant changes even those files.

Comment (dimpase) [reply configure]: using newer autotools fixed this by itself.

File: aclocal.m4 (hunk 1 above)
Comment (dimpase) [reply aclocal.m4]: ok, all tidy now, kept the old aclocal.

**Comment by jdemeyer** [comment]
This needs to be rebased to latest master.

**Comment by jdemeyer** [comment]
Or maybe it should be closed because of #13773?

**Comment by dimpase** [comment]
https://bugs.python.org/msg346049

**Comment by ned-deily** [comment]
Thanks again for the PR but, as discussed on the bug tracker, it doesn't cover all of the necessary use cases so we have adopted a different solution.
